domain/application/state/peer_relation.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+	"time"
+
+	"github.com/canonical/sqlair"
+
+	coreapplication "github.com/juju/juju/core/application"
+	corerelation "github.com/juju/juju/core/relation"
+	corestatus "github.com/juju/juju/core/status"
+	"github.com/juju/juju/internal/errors"
+)
+
+// getPeerEndpoints retrieves a list of peer endpoint for the given
+// application UUID from the database.
+func (st *State) getPeerEndpoints(ctx context.Context, tx *sqlair.TX, uuid coreapplication.ID) ([]peerEndpoint, error) {
+	type application dbUUID
+	app := application{UUID: uuid.String()}
+
+	stmt, err := st.Prepare(`
+SELECT 
+    ae.uuid AS &peerEndpoint.uuid,
+    cr.name AS &peerEndpoint.name
+FROM application_endpoint ae
+JOIN charm_relation cr ON cr.uuid = ae.charm_relation_uuid
+WHERE ae.application_uuid = $application.uuid
+AND cr.role_id = 2 -- Peer role`, app,
```

Comment:
We have the `charm_relation_role` table, let's use it rather than hard code a "random" number.

```suggestion
JOIN     charm_relation_role AS cmr ON cr.role_id = cmr.id 
WHERE ae.application_uuid = $application.uuid
AND      cmr.name = 'peer'
`, app,
```